src/main/java/com/coveo/configuration/parameterstore/strategy/MultiRegionParameterStorePropertySourceEnvironmentPostProcessStrategy.java

```diff
@@ -0,0 +1,62 @@
+package com.coveo.configuration.parameterstore.strategy;
+
+import static com.coveo.configuration.parameterstore.ParameterStorePropertySourceConfigurationProperty.HALT_BOOT;
+import static com.coveo.configuration.parameterstore.ParameterStorePropertySourceConfigurationProperty.SSM_CLIENT_SIGNING_REGIONS;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.ListIterator;
+
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.util.ObjectUtils;
+
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagement;
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClientBuilder;
+
+import com.coveo.configuration.parameterstore.ParameterStorePropertySource;
+import com.coveo.configuration.parameterstore.ParameterStoreSource;
+
+public class MultiRegionParameterStorePropertySourceEnvironmentPostProcessStrategy
+        implements ParameterStorePropertySourceEnvironmentPostProcessStrategy
+{
+    private static final String PARAMETER_STORE_PROPERTY_SOURCE_NAME = "MultiRegionAWSParameterStorePropertySource_";
+
+    @Override
+    public void postProcess(ConfigurableEnvironment environment)
+    {
+        boolean haltBoot = environment.getProperty(HALT_BOOT, Boolean.class, Boolean.FALSE);
+
+        LinkedList<String> signingRegions = getSigningRegions(environment);
+        ListIterator<String> regionsIterator = signingRegions.listIterator(signingRegions.size());
+
+        // To keep the order of precedence, we iterate from the last one to the first one because of the method 'addFirst'.
```

Comment:
While you're at it, explain why you can't use the addLast method, it's clear for me, but my not be for newcomers.